src/address.rs

```diff
@@ -33,7 +33,9 @@ impl Address {
         }
 
         Ok(Address::new(
-            Ipv4Addr::from(u32::from_be(addr.host)),
+            // use native byte order here, Enet already guarantees network byte order, so don't
+            // change it.
+            Ipv4Addr::from(addr.host.to_ne_bytes()),
             addr.port,
         ))
```

Comment:
I think this should just call `from_enet_address()`, so the logic isn't duplicated.